src/components/filters/Date/index.tsx

```diff
@@ -94,24 +103,18 @@ function DateFilter({ disabled, header, selectableTableStoreName }: Props) {
                     horizontal: 'left',
                 }}
             >
-                <MenuItem onClick={() => handlers.setFilter('today')}>
-                    <FormattedMessage id="filter.time.today" />
-                </MenuItem>
-                <MenuItem onClick={() => handlers.setFilter('yesterday')}>
-                    <FormattedMessage id="filter.time.yesterday" />
-                </MenuItem>
-                <MenuItem onClick={() => handlers.setFilter('thisWeek')}>
-                    <FormattedMessage id="filter.time.thisWeek" />
-                </MenuItem>
-                <MenuItem onClick={() => handlers.setFilter('lastWeek')}>
-                    <FormattedMessage id="filter.time.lastWeek" />
-                </MenuItem>
-                <MenuItem onClick={() => handlers.setFilter('thisMonth')}>
-                    <FormattedMessage id="filter.time.thisMonth" />
-                </MenuItem>
-                <MenuItem onClick={() => handlers.setFilter('lastMonth')}>
-                    <FormattedMessage id="filter.time.lastMonth" />
-                </MenuItem>
+                <ListItem>
+                    <Typography>
+                        <FormattedMessage id="filter.time.label" />
+                    </Typography>
+                </ListItem>
+                <Divider />
+                <DateFilterOption {...optionProps} option="today" />
+                <DateFilterOption {...optionProps} option="yesterday" />
+                <DateFilterOption {...optionProps} option="thisWeek" />
+                <DateFilterOption {...optionProps} option="lastWeek" />
+                <DateFilterOption {...optionProps} option="thisMonth" />
+                <DateFilterOption {...optionProps} option="lastMonth" />
```

Comment:
Made this reusable to prevent copy/paste issues

Reply:
yeah - eventually this'll all live in an array of some sort. Mainly after we move more stuff into store/context

The extra component is kind of a feature and not a bug. The goal will be to move more stuff into stores/context and then all the options don't need to re-render when something changes and are only involved with their own "state".